src/Sentry.Profiling/SampleProfilerSession.cs

```diff
@@ -50,17 +51,60 @@ private SampleProfilerSession(SentryStopwatch stopwatch, EventPipeSession sessio
 
     public TraceLog TraceLog => _eventSource.TraceLog;
 
+    internal static bool ThrowOnNextStartupForTests = false;
+
     public static SampleProfilerSession StartNew(IDiagnosticLogger? logger = null)
     {
-        var client = new DiagnosticsClient(Process.GetCurrentProcess().Id);
-        var session = client.StartEventPipeSession(Providers, requestRundown: false, CircularBufferMB);
-        var stopWatch = SentryStopwatch.StartNew();
-        var eventSource = TraceLog.CreateFromEventPipeSession(session, TraceLog.EventPipeRundownConfiguration.Enable(client));
+        try
+        {
+            var client = new DiagnosticsClient(Process.GetCurrentProcess().Id);
+
+            if (ThrowOnNextStartupForTests)
+            {
+                ThrowOnNextStartupForTests = false;
+                throw new Exception("Test exception");
+            }
 
-        // Process() blocks until the session is stopped so we need to run it on a separate thread.
-        Task.Factory.StartNew(eventSource.Process, TaskCreationOptions.LongRunning);
+            // Note: StartEventPipeSession() can time out after 30 seconds on resource constrained systems.
+            // See https://github.com/dotnet/diagnostics/blob/991c78895323a953008e15fe34b736c03706afda/src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcClient.cs#L40C52-L40C52
+            var session = client.StartEventPipeSession(Providers, requestRundown: false, CircularBufferMB);
 
-        return new SampleProfilerSession(stopWatch, session, eventSource, logger);
+            var stopWatch = SentryStopwatch.StartNew();
+            var eventSource = TraceLog.CreateFromEventPipeSession(session, TraceLog.EventPipeRundownConfiguration.Enable(client));
+
+            // Process() blocks until the session is stopped so we need to run it on a separate thread.
+            Task.Factory.StartNew(eventSource.Process, TaskCreationOptions.LongRunning)
+                .ContinueWith(_ =>
+                {
+                    if (_.Exception?.InnerException is { } e)
+                    {
+                        logger?.LogWarning("Error during sampler profiler EventPipeSession processing.", e);
```

Comment:
Will conflict now I imagine, since we merged ur PR reordering

Reply:
Actually, I should have looked for usages on the other PR too... All the usages passing an exception got picked up by the generic method so they didn't fail. I've fixed it in this branch with hope it gets merged soon also those fixes are unrelated. If it doesn't get merged, I can cherry-pick the commit to a new PR.